lib/error.ts

```diff
@@ -6,6 +6,8 @@ export enum ErrorCodes {
   NOT_FOUND = 4,
   ALREADY_EXISTS = 5,
   ENCODER = 6,
+  DECODER = 7,
```

Comment:
I know it sounds silly, but let's use the encoder type for both encoding and decoding errors